### PR TITLE
Connecticut Pension or Annuity Income Subtraction

### LINKED ISSUE #3011: Conneticut Pension or Annuity Income Subtraction
Reference:
https://www.cga.ct.gov/current/pub/chap_229.htm#sec_12-701a

Section:
 (20), (B), (xxi)

### PR DESCRIPTION
**Pull request by ThapeloMasasa**
Fixes #3011

### Summary
📱🌎🎨

Add a new policy parameter `ubi_amount` to the US tax and benefit system. Use this parameter to implement a simple universal basic income (UBI) reform in `reform_1.yaml`. Update the web app to display the UBI amount and the net income effect of the reform.

> _`is_valid` checks_
> _refactored for clarity_
> _autumn leaves falling_

### Walkthrough
*  Add a new parameter `child_allowance.amount` to the `parameters` folder, which defines the monthly amount of the child allowance benefit per child ()

### CODE REVIEW

File: changelog_entry.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
+  changes:
+    added:
+    - Conneticut Pension or Annuity Income Subtraction
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:4]:
```suggestion
    - Connecticut pension or annuity income subtraction.
```

File: policyengine_us/parameters/gov/states/ct/tax/income/subtractions/pensions_or_annuity/rate.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+description: Connecticut provides this fraction of annuity and pension income under the annuity and pension subtraction.
+values:
+  2019-01-01: 0.14 
+  2020-01-01: 0.28
+  2021-01-01: 0.42
+  2022-01-01: 1.00
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ct/tax/income/subtractions/pensions_or_annuity/rate.yaml:6]:
```suggestion
  2022-01-01: 1
```

File: policyengine_us/parameters/gov/states/ct/tax/income/subtractions/pensions_or_annuity/rate.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+metadata:
+  unit: /1
+  label: Connecticut annuity and pension subtraction income percentage
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ct/tax/income/subtractions/pensions_or_annuity/rate.yaml:9]:
```suggestion
  label: Connecticut annuity and pension subtraction income rate
```

**Review by PavelMakarchuk — APPROVED** [review]

**Comment by PavelMakarchuk** [comment]
@MaxGhenis we have a general Subtractions PR where this will be included, hence it is without the income tree inclusion

File: policyengine_us/parameters/gov/states/ct/tax/income/subtractions/pensions_or_annuity/rate.yaml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+description: Connecticut provides this fraction of annuity and pension income under the annuity and pension subtraction.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/ct/tax/income/subtractions/pensions_or_annuity/rate.yaml:1]:
```suggestion
description: Connecticut subtracts this fraction of annuity and pension income from adjusted gross income.
```

File: policyengine_us/parameters/gov/states/ct/tax/income/subtractions/pensions_or_annuity/income_threshold.yaml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+metadata:
+  period: year
+  unit: currency-USD
+  label: Connecticut annuity and pension subtraction income threshold
+  breakdown:
+    - filing_status
+  reference:
+    - title: Connecticut General Statutes, Chapter 229, (20), (B), (xxi) 
+      href: https://www.cga.ct.gov/current/pub/chap_229.htm#sec_12-701a
+    - title: 2022 Form CT-1040 Connecticut Resident Income Tax Return Instructions
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/ct/tax/income/subtractions/pensions_or_annuity/income_threshold.yaml:11]:
```suggestion
    - title: 2022 Form CT-1040 Connecticut Resident Income Tax Return Instructions, Line 48b
```

File: policyengine_us/parameters/gov/states/ct/tax/income/subtractions/pensions_or_annuity/rate.yaml [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+values:
+  2019-01-01: 0.14 
+  2020-01-01: 0.28
+  2021-01-01: 0.42
+  2022-01-01: 1
+metadata:
+  unit: /1
+  label: Connecticut annuity and pension subtraction income rate
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/ct/tax/income/subtractions/pensions_or_annuity/rate.yaml:9]:
```suggestion
  label: Connecticut annuity and pension subtraction rate
```

File: policyengine_us/parameters/gov/states/ct/tax/income/subtractions/pensions_or_annuity/income_threshold.yaml
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/ct/tax/income/subtractions/pensions_or_annuity/income_threshold.yaml:1]: rename income_limit.yaml

File: policyengine_us/parameters/gov/states/ct/tax/income/subtractions/pensions_or_annuity/income_threshold.yaml [hunk 7]
```diff
@@ -0,0 +1,22 @@
+description: Connecticut limits the annuity and pension subtraction to filers with adjusted gross income below this threshold, based on filing status.
+metadata:
+  period: year
+  unit: currency-USD
+  label: Connecticut annuity and pension subtraction income threshold
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/ct/tax/income/subtractions/pensions_or_annuity/income_threshold.yaml:5]:
```suggestion
  label: Connecticut annuity and pension subtraction income limit
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

**Review by MaxGhenis — APPROVED** [review]

File: policyengine_us/parameters/gov/states/ct/tax/income/subtractions/pensions_or_annuity/rate.yaml [hunk 8]
```diff
@@ -0,0 +1,14 @@
+description:  Connecticut subtracts this fraction of annuity and pension income from adjusted gross income.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/ct/tax/income/subtractions/pensions_or_annuity/rate.yaml:1]:
```suggestion
description: Connecticut subtracts this fraction of annuity and pension income from adjusted gross income.
```

File: policyengine_us/parameters/gov/states/ct/tax/income/subtractions/pensions_or_annuity/rate.yaml [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+values:
+  2019-01-01: 0.14 
+  2020-01-01: 0.28
+  2021-01-01: 0.42
+  2022-01-01: 1
+metadata:
+  unit: /1
+  label:  Connecticut annuity and pension subtraction rate
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/ct/tax/income/subtractions/pensions_or_annuity/rate.yaml:9]:
```suggestion
  label: Connecticut annuity and pension subtraction rate
```

File: policyengine_us/tests/policy/baseline/gov/states/ct/tax/income/subtractions/ct_pension_annuity_eligible.yaml [hunk 10]
```diff
@@ -0,0 +1,35 @@
+- name: Eligible
```
Comment (MaxGhenis) [inline policyengine_us/tests/policy/baseline/gov/states/ct/tax/income/subtractions/ct_pension_annuity_eligible.yaml:1]: Could you make the test names more descriptive?

**Review by MaxGhenis — CHANGES_REQUESTED** [review]
Please also git pull upstream master to pass the test

File: policyengine_us/variables/gov/states/ct/tax/income/subtractions/pension_annuity/ct_pension_annuity_subtraction.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ct/tax/income/subtractions/pension_annuity/ct_pension_annuity_subtraction.py:1]: add this variable to the list of subtractions in the subtractions.yaml

**Pull request merged**